### ISSUE TITLE
Questions about decomp_century code

### ISSUE THREAD

**Issue by bpbond** [PNNL-TES/millenial#7]
The big one is: why are all the lines commented out?

Secondarily, I don't get why `f_DOC_ATM` seems to be passed in and not computed inside the routine, as all the other fluxes (`f_ACTIVE_DOC` etc.) are.

Thanks!

**Comment by bpbond** [comment]
> f_DOC_ATM seems to be passed in and not computed inside the routine

Ah, it's supposed to be computed in line 956, but there's a typo--`f_DOC_leaching` is repeated twice:
```fortran
 949 	!~ ! DOC -> leaching
 950 	!~ if (DOC > 0._r8) then
 951         !~ f_DOC_leaching = DOC * 0.0001 * (DOC/(DOC + 100)) / dt * t_scalar * w_scalar
 952 	!~ end if
 953 
 954 	!~ ! DOC -> ATM
 955 	!~ if (DOC > 0._r8) then
 956         !~ f_DOC_leaching = DOC * 0.005 / dt * t_scalar * w_scalar
 957 	!~ end if
 958 	
```

**Comment by bpbond** [comment]
Final question: all the fluxes get divided by `dt` (in lines 951, 956, ...995 in `main.F90`) and then multiplied by it (999-1002). As a result `timestep` has no effect. Is this intended?

**Comment by email-clm** [comment]
Hi Ben,

That is right for the dt in the model. Eventually it is a better way for
modeling ecological processes, and it is taken in the CLM (half-hour time
step, but some processes on second). It does take consideration of the time
step, particularly for some processes with low donor pools. With low donor
pool size, the process rate multiple time step will led to negative pool
size. Generally speaking, it is a good approach to resolve inconsistencies
between all nonlinear ecological processes. My understanding might be
incomplete, it might have the advantages as well. The bad thing is its
time-consuming for computation.

Best,
Xiaofeng

**Comment by bpbond** [comment]
Yeah, it seems like using a numerical integration method would be much more robust. Something for the to-do list.

The `f_DOC_ATM` / `f_DOC_leaching` thing above is a genuine bug, I assume.

Thanks!

**Comment by email-clm** [comment]
Yes, the numerical integration method is much better. Yet for most
ecological models, rather than geochemical models, the time steps are
fixed. Generally speaking, finer time step is better for model performance,
while computational cost is much much high.

The issue for f_DOC_ATM and f_DOC_leaching is not a bug, since they are
both treated as DOC loss, from two different mechanisms. For the current
model version, we focused on C pool, rather than C flux. Both f_DOC_ATM and
f_DOC_leaching are for same process, they were combined into one process.
You can put a note for that, and separate in your own code. Probably set
leaching parameter as zero for current stage to be consistent with our
model, it should be modified (calibrated) if you want to separate the
different pathways for C loss.

Best,
Xiaofeng

**Comment by bpbond** [comment]
>Both f_DOC_ATM and f_DOC_leaching are for same process, they were combined into one process.

So the second `f_DOC_leaching` calculation, in line 956, which overwrites the results of the first calculation, is intended to calculate _both_ leaching and the DOC loss to the atmosphere? Just making sure I understand–thanks.
